components/prism-bash.js

```diff
@@ -19,66 +29,168 @@
 					punctuation: /\(\(?|\)\)?|,|;/
 				}
 			},
-			// Command Substitution
+			// [1]: Command Substitution
 			{
-				pattern: /\$\([^)]+\)|`[^`]+`/,
+				pattern: /\$\((?:\([^)]+\)|[^()])+\)|`[^`]+`/,
 				greedy: true,
 				inside: {
 					variable: /^\$\(|^`|\)$|`$/
 				}
 			},
-			/\$(?:[\w#?*!@]+|\{[^}]+\})/i
-		]
+			// [2]: Brace expansion
+			{
+				pattern: /\$\{[^}]+\}/,
+				greedy: true,
+				inside: {
+					operator: /:[-=?+]?|[!\/]|##?|%%?|\^\^?|,,?/,
+					punctuation: /[\[\]]/,
+					environment: {
+						pattern: new RegExp("(\\{)(?:" + envVarList + ")(?=$|\\}|\\W)"),
+						lookbehind: true,
+						alias: 'constant'
+					}
+				}
+			},
+			/\$(?:\w+|[#?*!@$])/
+		],
+		// man echo / printf, + \"
+		entity: /\\([abceEfnrtv\\"]|O?[0-7]{1,3}|x[0-9a-fA-F]{1,2}|u[0-9a-fA-F]{4}|U[0-9a-fA-F]{8})/
 	};
 
 	Prism.languages.bash = {
-		'shebang': {
-			pattern: /^#!\s*\/bin\/bash|^#!\s*\/bin\/sh/,
+		shebang: {
+			pattern: /^#!\s*\/.*/,
 			alias: 'important'
 		},
-		'comment': {
-			pattern: /(^|[^"{\\])#.*/,
+		comment: {
+			pattern: /(^|[^"{\\$])#.*/,
 			lookbehind: true
 		},
-		'string': [
-			//Support for Here-Documents https://en.wikipedia.org/wiki/Here_document
+		'function-name': [
+			// a) function foo {
+			// b) foo() {
+			// c) function foo() {
+			// but not “foo {”
 			{
-				pattern: /((?:^|[^<])<<\s*)["']?(\w+?)["']?\s*\r?\n(?:[\s\S])*?\r?\n\2/,
+				// a) and c)
+				pattern: /((^|\W)function\s+)\w+(?=(\s*\(\s*\))?\s*\{)/,
```

Comment:
1. Please make the non-lookbehind capturing groups non-capturing.
2. You can replace `(^|\W)` with a simple `\b`.